--- mtd-utils-9f10713.orig/Makefile	2014-04-07 13:01:36.000000000 +0200
+++ mtd-utils-9f10713/Makefile	2014-08-25 10:43:06.000000000 +0200
@@ -18,7 +18,7 @@
 
 MTD_BINS = \
 	ftl_format flash_erase nanddump doc_loadbios \
-	ftl_check mkfs.jffs2 flash_lock flash_unlock \
+	ftl_check flash_lock flash_unlock \
 	flash_otp_info flash_otp_dump flash_otp_lock flash_otp_write \
 	mtd_debug flashcp nandwrite nandtest \
 	jffs2dump \
@@ -31,7 +31,7 @@
 	ubidetach ubinize ubiformat ubirename mtdinfo ubirsvol ubiblock
 
 BINS = $(MTD_BINS)
-BINS += mkfs.ubifs/mkfs.ubifs
+#BINS += mkfs.ubifs/mkfs.ubifs
 BINS += $(addprefix ubi-utils/,$(UBI_BINS))
 SCRIPTS = flash_eraseall
 
